Source/DataSources/KmlDataSource.js

```diff
@@ -255,6 +255,33 @@ define([
         return deferred.promise;
     }
 
+    function insertNamespaces(text) {
+        var namespaceMap = {
+            xsi : 'http://www.w3.org/2001/XMLSchema-instance'
+        };
+        var firstPart, lastPart, reg, declaration;
+
+        for (var key in namespaceMap) {
+            if (namespaceMap.hasOwnProperty(key)) {
+                reg = RegExp('[<| ]' + key + ':');
```

Comment:
`[ ]` specifies a character group where it represents any of the characters in it. This one will match `<xsi:`, ` xsi:` or `|xsi:`. I think you were trying to use the OR syntax that you need in capture group (represented with `( )`). So it can either be `[< ]` or `(<| )`. I think just removing the `|` from your code and going with the character group is the way to go as it doesn't have the overhead of capturing.